File: csharp/raindrops/Raindrops.cs
using System;

public static class Raindrops
{
    public static string Convert(int number)
    {
        string sound = "";

        if (IsDivisible(number, 3))
            sound += "Pling";
        if (IsDivisible(number, 5))
            sound += "Plang";
        if (IsDivisible(number, 7))
            sound += "Plong";
        if (sound == "")
            sound += number.ToString();

        return sound;
    }

    private static bool IsDivisible(int x, int y)
    {
        return x % y == 0;
    }
}

File: lua/raindrops/raindrops.lua
local function is_divisible(x, y)
  return x % y == 0
end

return function(n)
  local output = ""

  if is_divisible(n, 3) then output = output .. "Pling" end
  if is_divisible(n, 5) then output = output .. "Plang" end
  if is_divisible(n, 7) then output = output .. "Plong" end

  if #output == 0 then output = output .. tostring(n) end

  return output
end
